lib/spack/spack/cmd/buildcache.py

```diff
@@ -265,56 +274,106 @@ def setup_parser(subparser):
         default=None,
         help="A quoted glob pattern identifying copy manifest files",
     )
-    source = sync.add_mutually_exclusive_group(required=False)
```

Comment:
@scottwittenburg do you recall why this was optional?

Reply:
The purpose of it is to communicate the specs that needed rebuliding in the pipeline and should be copied up to the root of the mirror (from the stack-specific mirrors).  While we always do that in our protected pipelines, it's not a feature I would expect gets used by others who might use the `spack ci` commands.  So I made it optional.

Reply:
I still don't get it :p it has to be synced from mirror/buildcache A to B, so you need to tell Spack what A and B are? Right now you can also not specify A and B?

Reply:
The manifest files are json files that say exactly what subset of specs to copy and include `src` and `dest` urls for each spec.  In the case of spack pipelines, the manifest files are produced by the pipeline generation job, and they're passed (via artifacts) to `protected-publish` job.  Here's an [example](https://gitlab.spack.io/spack/spack/-/jobs/5128518/artifacts/browse/jobs_scratch_dir/specs_to_copy/).

Reply:
Okay, well, whatever. I made it optional again.